### PR TITLE
docs: added docs page about checking permissions

### PR DESCRIPTION
**Pull request by Commandserver**
This PR adds a new docs page about permissions and makes the glossar appear at the top of the "related pages".

- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.
- [x] I agree to the terms of the [DCO (Developer Certificate of Origin)]((https://dpp.dev/coding-standards.html))

### CODE REVIEW

File: docpages/example_programs/the_basics/checking-member-permissions.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+\page checking-member-permissions Checking permissions
+
+The easiest way to check member permissions are either the dpp::channel::get_user_permissions or dpp::guild::permission_overwrites method. Both do the same under the hood.
+
+Of course most people do just iterate over the roles of a member to check they for a permission.
```
Comment (braindigitalis) [inline docpages/example_programs/the_basics/checking-member-permissions.md:5]: to check they for -> to check for

File: docpages/01_disdppgloss.md [hunk 2]
```diff
@@ -1,51 +1,51 @@
-## A Glossary of Common Discord Terms
```
Comment (braindigitalis) [inline docpages/01_disdppgloss.md:1]: not sure this should come before the faq, maybe after it?

Comment (Commandserver) [reply docpages/01_disdppgloss.md:1]: okay. ill put it after it

Comment (Commandserver) [reply docpages/01_disdppgloss.md]: had to rename all other pages 🤦

**Review by Mishura4 — APPROVED** [review]

**Comment by Mishura4** [comment]
I would put it in `Interactions and Components` tbh!
Though the basics could work

**Review by braindigitalis — APPROVED** [review]

**Review by Mishura4 — APPROVED** [review]

**Pull request merged**